packages/babel-parser/src/parser/lval.js

```diff
@@ -60,7 +65,12 @@ export default class LValParser extends NodeUtils {
           ) {
             const prop = node.properties[i];
             const isLast = i === last;
-            this.toAssignableObjectExpressionProp(prop, isBinding, isLast);
+            this.toAssignableObjectExpressionProp(
+              prop,
+              isBinding,
+              isLast,
+              allowAssign,
```

Comment:
Why is it passed down? Inside an object property, assignments are always allowed (for default props)

Reply:
The original approach needed to pass it down to "remember" that there had been a parenthesis somewhere up the stack - to invalidate expressions such as `[({ a: [b = 2]})] = t`. It's become more clear to me now that the presence of any parenthesis in the left-hand side of an assignment expression is the problem so I'll clean this up.